card/lib/card/view.rb

```diff
@@ -18,9 +18,16 @@ class View
 
     # @return [Symbol]
     def self.canonicalize view
-      return if view.blank? # error?
-      view.to_sym
-#      view.to_viewname.key.to_sym
+      case view
+      when blank? # error?
+        nil
+      when Symbol
+        view
+      when String
+        view.gsub(/\W/, "").to_sym
```

Comment:
you could make the case that Viewname was better than this because it cached this gsub.  But it was a whole lot of code for that!  We can probably do a little per-process cache here.

Reply:
Instead of this we could fix the bad view call in wikirate. We don't have to correct views if somebody puts wrong characters in it. Just a "view doesn't exist" is also fine.